cpmpy/expressions/globalconstraints.py

```diff
@@ -378,11 +383,17 @@ def value(self):
     def decompose_comparison(self, cpm_op, cpm_rhs):
         """
         Decomposition if it's part of a comparison
+        Returns two lists of constraints:
+            1) constraint replacing node in expression tree
+            2) constraints which should be true at toplevel
         """
         from .python_builtins import any, all
+        if cpm_op == "==": # can avoid creating aux var
+            return [any(x == cpm_rhs for x in self.args), all(x >= cpm_rhs for x in self.args)]
```

Comment:
please make this always return a tuple (expr, expr) instead of [expr, expr], so we can actually check that it is a tuple and not one list of constraints.